x-pack/plugins/security_solution/public/detections/pages/detection_engine/chart_panels/alerts_local_storage/index.tsx

```diff
@@ -0,0 +1,125 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocalStorage } from '../../../../../common/components/local_storage';
+import {
+  getSettingKey,
+  isDefaultWhenEmptyString,
+} from '../../../../../common/components/local_storage/helpers';
+import { APP_ID } from '../../../../../../common/constants';
+import {
+  ALERTS_PAGE,
+  ALERT_VIEW_SELECTION_SETTING_NAME,
+  TABLE_CATEGORY,
+  EXPAND_SETTING_NAME,
+  TREEMAP_CATEGORY,
+  STACK_BY_0_SETTING_NAME,
+  STACK_BY_1_SETTING_NAME,
+  STACK_BY_SETTING_NAME,
+  TREND_CHART_CATEGORY,
+  VIEW_CATEGORY,
+} from './constants';
+import {
+  DEFAULT_STACK_BY_FIELD,
+  DEFAULT_STACK_BY_FIELD1,
+} from '../../../../components/alerts_kpis/common/config';
+import type { AlertsSettings } from './types';
+import type { AlertViewSelection } from '../chart_select/helpers';
+import { TREND_ID } from '../chart_select/helpers';
+
+export const useAlertsLocalStorage = (): AlertsSettings => {
+  const [alertViewSelection, setAlertViewSelection] = useLocalStorage<AlertViewSelection>({
+    defaultValue: TREND_ID,
+    key: getSettingKey({
+      category: VIEW_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: ALERT_VIEW_SELECTION_SETTING_NAME,
+    }),
+    plugin: APP_ID,
+    isInvalidDefault: isDefaultWhenEmptyString,
+  });
+
+  const [isTreemapPanelExpanded, setIsTreemapPanelExpanded] = useLocalStorage<boolean>({
+    defaultValue: true,
+    key: getSettingKey({
+      category: TREEMAP_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: EXPAND_SETTING_NAME,
+    }),
+    plugin: APP_ID,
+  });
+
+  const [riskChartStackBy0, setRiskChartStackBy0] = useLocalStorage<string>({
+    defaultValue: DEFAULT_STACK_BY_FIELD,
+    key: getSettingKey({
+      category: TREEMAP_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: STACK_BY_0_SETTING_NAME,
+    }),
+    plugin: APP_ID,
+    isInvalidDefault: isDefaultWhenEmptyString,
+  });
+
+  const [riskChartStackBy1, setRiskChartStackBy1] = useLocalStorage<string | undefined>({
+    defaultValue: DEFAULT_STACK_BY_FIELD1,
+    key: getSettingKey({
+      category: TREEMAP_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: STACK_BY_1_SETTING_NAME,
+    }),
+    plugin: APP_ID,
+  });
+
+  const [countTableStackBy0, setCountTableStackBy0] = useLocalStorage<string>({
+    defaultValue: DEFAULT_STACK_BY_FIELD,
+    key: getSettingKey({
+      category: TABLE_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: STACK_BY_0_SETTING_NAME,
+    }),
+    plugin: APP_ID,
+    isInvalidDefault: isDefaultWhenEmptyString,
+  });
+
+  const [countTableStackBy1, setCountTableStackBy1] = useLocalStorage<string | undefined>({
+    defaultValue: DEFAULT_STACK_BY_FIELD1,
+    key: getSettingKey({
+      category: TABLE_CATEGORY,
+      page: ALERTS_PAGE,
+      setting: STACK_BY_1_SETTING_NAME,
+    }),
+    plugin: APP_ID,
```

Comment:
Yea, I'd say let's default plugin `APP_ID` in the actual `useLocalStorage`. It can be in a follow up PR for sure if it'll be too much of a headache here with the tests 😂

Reply:
> Yea, I'd say let's default plugin `APP_ID` in the actual `useLocalStorage`. It can be in a follow up PR for sure if it'll be too much of a headache here with the tests 😂

done! 😄